portfolio-project-matching-app/backend/dao.js

```diff
@@ -356,9 +449,11 @@ const deleteAssociation = async (coll, id1, id2) => {
     // parse association collection name to get individual collection names
     const [coll1, coll2] = coll.split('_');
     // get document snapshots for invalid input handling
-    const id1Snap = await getDocSnapshotById(coll1, id1);
-    const id2Snap = await getDocSnapshotById(coll2, id2);
-    const collSnap = await getDocSnapshotById(coll, `${id1}_${id2}`);
+    const [id1Snap, id2Snap, collSnap] = await Promise.all([
+        getDocSnapshotById(coll1, id1),
+        getDocSnapshotById(coll2, id2),
+        getDocSnapshotById(coll, `${id1}_${id2}`)
+    ]);
```

Comment:
Great call by Josh on these types of improvements. No need to await each other's resolve.. let them do their work and move forward when they all finish asynchronously.